### PR TITLE
[#30] Add enclave selection in Jenkinsfiles

### LINKED ISSUE #30: Enclave selection capabilities in Jenkinsfile
At the moment legion plugin does not provide capability to select enclave/namespace to use for training and execution of a model. 

We need to introduce additional parameters for legion.pod() and legion.deploy() routine to access use namespace.

### PR DESCRIPTION
**Pull request by kirillmakhonin**
This closes #30

### CODE REVIEW

**Comment by alexander-semenets** [comment]
ok to test

**Comment by alexander-semenets** [comment]
test this please

**Comment by alexander-semenets** [comment]
LGTM, but should be rebased and retested again before merge

**Comment by kirillmakhonin** [comment]
test this please

**Comment by kirillmakhonin** [comment]
test this please

**Comment by kirillmakhonin** [comment]
test this please

**Comment by kirillmakhonin** [comment]
ok to test

**Comment by kirillmakhonin** [comment]
ok to test

**Comment by kirillmakhonin** [comment]
test this please

File: k8s/jenkins/legion-jenkins-plugin/src/main/resources/legion.groovy [hunk 1]
```diff
@@ -48,28 +66,49 @@ def modelFileName() {
...
     }
+    else
+        println "Using enclave ${enclave}"
 
-    image = podParams.get('image', getDefaultImageName())
-    cpu = podParams.get('cpu', '330m')
-    ram = podParams.get('ram', '4Gi')
-
+    // Create list of env variables that should be passed to slave
     envToPass = [
```
Comment (dsuslov) [inline k8s/jenkins/legion-jenkins-plugin/src/main/resources/legion.groovy]: @kirillmakhonin the list of env vars should be defined in configuration file but not hardcoded in groovy lib script

File: k8s/jenkins/legion-jenkins-plugin/src/main/resources/legion.groovy [hunk 2]
```diff
@@ -80,11 +119,21 @@ def pod(Map podParams=null, Closure body) {
...
 
-    envVars = envToPass.collect({ name -> envVar(key: name, value: System.getenv(name)) })
-    envVars << envVar(key: 'ENCLAVE_DEPLOYMENT_PREFIX', value: "${env.ENCLAVE_DEPLOYMENT_PREFIX}")
-    envVars << envVar(key: 'MODEL_SERVER_URL', value: "http://${env.ENCLAVE_DEPLOYMENT_PREFIX}${params.Enclave}-edge.${params.Enclave}")
-    envVars << envVar(key: 'EDI_URL', value: "http://${env.ENCLAVE_DEPLOYMENT_PREFIX}${params.Enclave}-edi.${params.Enclave}")
+    // Collect list of ENV variables to pass to slave instance
+    envVarsList = envToPass.collect({ name -> envVar(key: name, value: System.getenv(name)) })
+    envVarsList << envVar(key: 'ENCLAVE', value: "${enclave}")
+    envVarsList << envVar(key: 'ENCLAVE_DEPLOYMENT_PREFIX', value: "${env.ENCLAVE_DEPLOYMENT_PREFIX}")
+    envVarsList << envVar(key: 'MODEL_SERVER_URL', value: "http://${env.ENCLAVE_DEPLOYMENT_PREFIX}${enclave}-edge.${enclave}")
```
Comment (dsuslov) [inline k8s/jenkins/legion-jenkins-plugin/src/main/resources/legion.groovy]: @kirillmakhonin urls templates should be defined in configuration but not hardcoded in groovy pipeline lib script

File: k8s/jenkins/legion-jenkins-plugin/src/main/resources/legion.groovy [hunk 3]
```diff
@@ -16,14 +16,29 @@
  *   limitations under the License.
  */
 
+class Globals {
```
Comment (dsuslov) [inline k8s/jenkins/legion-jenkins-plugin/src/main/resources/legion.groovy]: @kirillmakhonin what is the purpose of Globals if the same global keys are overridden in several different lib method? What we will get if these methods are executed in parallel ?  Such global vars should be defined once in some init() method that is executed just once as a first step of pipeline script.

**Comment by dsuslov** [comment]
it is not actual since CI server will be moved out of Legion cluster. See #616

**Pull request closed without merging**